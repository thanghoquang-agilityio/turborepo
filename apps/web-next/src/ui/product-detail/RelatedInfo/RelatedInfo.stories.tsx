import type { Meta, StoryObj } from '@storybook/react'

import RelatedInfo from '.'

const meta = {
  title: 'Components/RelatedInfo',
  tags: ['autodocs'],
  component: RelatedInfo,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof RelatedInfo>

export default meta
type Story = StoryObj<typeof RelatedInfo>

export const RelatedInfoDefault: Story = {}
