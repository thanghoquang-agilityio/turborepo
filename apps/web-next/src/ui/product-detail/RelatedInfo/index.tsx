import { ReactNode, memo } from 'react'

import Link from 'next/link'

// Components
import { Text } from '@/components'
// Constants
import { borderStyle } from '@/constants'
import { ShoppingBagIcon, TruckIcon } from '@/icons'

interface RelatedInfoItem {
  icon: ReactNode
  title: string
  subTitle?: ReactNode
}

export const RELATED_INFO = [
  {
    icon: <TruckIcon />,
    title: 'Free Delivery',
    subTitle: (
      <Link href='#'>
        <Text
          variant='secondary'
          size='sm'
          type='link'
          className='overflow-visible whitespace-pre-wrap leading-6 text-foreground-100 underline-offset-4 sm:leading-4'
        >
          Enter your Postal code for Delivery Availability
        </Text>
      </Link>
    ),
  },
  {
    icon: <ShoppingBagIcon />,
    title: 'Return Delivery',
    subTitle: (
      <Text
        variant='secondary'
        size='sm'
        className='overflow-visible whitespace-pre-wrap leading-6 text-foreground-100 sm:leading-4'
      >
        Free 30 days Delivery Return.&nbsp;
        <Link href='#'>
          <span className='underline underline-offset-4'>Details</span>
        </Link>
      </Text>
    ),
  },
]

const RelatedInfo = memo(
  ({ items = RELATED_INFO }: { items?: RelatedInfoItem[] }) => (
    <div
      className={`flex w-full flex-col gap-10 rounded-[14px] border-small border-background-300 p-[17px]`}
    >
      {items.map(({ icon, title, subTitle }, index) => {
        const borderClass =
          index !== 0 ? `${borderStyle} before:top-[-20px]` : ''

        return (
          <div
            key={`related-info-item-${index}`}
            className={`relative flex gap-[14px] sm:h-[45px] ${borderClass}`}
          >
            <div className='h-6 w-6 text-danger-100'>{icon}</div>
            <div className='flex flex-col gap-[7px] overflow-hidden'>
              <Text className='text-[17px] font-bold leading-5 text-primary-100'>
                {title}
              </Text>
              {subTitle}
            </div>
          </div>
        )
      })}
    </div>
  )
)

RelatedInfo.displayName = 'RelatedInfo'
export default RelatedInfo
