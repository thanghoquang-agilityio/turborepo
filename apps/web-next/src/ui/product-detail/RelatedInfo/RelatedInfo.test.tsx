import { render } from '@testing-library/react'

import RelatedInfo, { RELATED_INFO } from '.'

describe('RelatedInfo', () => {
  const renderComponent = () => render(<RelatedInfo items={RELATED_INFO} />)

  it('should match snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
