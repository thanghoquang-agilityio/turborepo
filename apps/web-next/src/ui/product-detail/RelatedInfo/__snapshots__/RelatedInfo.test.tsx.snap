// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedInfo should match snapshot 1`] = `
<div>
  <div
    class="flex w-full flex-col gap-10 rounded-[14px] border-small border-background-300 p-[17px]"
  >
    <div
      class="relative flex gap-[14px] sm:h-[45px] "
    >
      <div
        class="h-6 w-6 text-danger-100"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25002 18.75C8.25002 19.1478 8.09198 19.5293 7.81068 19.8107C7.52937 20.092 7.14784 20.25 6.75002 20.25C6.35219 20.25 5.97066 20.092 5.68936 19.8107C5.40805 19.5293 5.25002 19.1478 5.25002 18.75M8.25002 18.75C8.25002 18.3522 8.09198 17.9706 7.81068 17.6893C7.52937 17.408 7.14784 17.25 6.75002 17.25C6.35219 17.25 5.97066 17.408 5.68936 17.6893C5.40805 17.9706 5.25002 18.3522 5.25002 18.75M8.25002 18.75H14.25M5.25002 18.75H3.37502C3.07665 18.75 2.7905 18.6315 2.57952 18.4205C2.36855 18.2095 2.25002 17.9234 2.25002 17.625V14.25M14.25 18.75H16.5M14.25 18.75V14.25M2.25002 14.25V6.61499C2.24843 6.34116 2.34848 6.07648 2.53081 5.87217C2.71313 5.66786 2.96477 5.53845 3.23702 5.50899C6.57006 5.16302 9.92997 5.16302 13.263 5.50899C13.828 5.56699 14.25 6.04699 14.25 6.61499V7.57299M2.25002 14.25H14.25M19.5 18.75C19.5 19.1478 19.342 19.5293 19.0607 19.8107C18.7794 20.092 18.3978 20.25 18 20.25C17.6022 20.25 17.2207 20.092 16.9394 19.8107C16.6581 19.5293 16.5 19.1478 16.5 18.75M19.5 18.75C19.5 18.3522 19.342 17.9706 19.0607 17.6893C18.7794 17.408 18.3978 17.25 18 17.25C17.6022 17.25 17.2207 17.408 16.9394 17.6893C16.6581 17.9706 16.5 18.3522 16.5 18.75M19.5 18.75H20.625C21.246 18.75 21.754 18.246 21.715 17.626C21.5141 14.3236 20.4019 11.1416 18.502 8.43299C18.3211 8.17927 18.0848 7.97001 17.8111 7.82102C17.5373 7.67203 17.2333 7.58722 16.922 7.57299H14.25M14.25 7.57299V14.25"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <div
        class="flex flex-col gap-[7px] overflow-hidden"
      >
        <p
          class="  no-underline text-ellipsis whitespace-nowrap overflow-hidden text-[17px] font-bold leading-5 text-primary-100"
        >
          Free Delivery
        </p>
        <a
          href="#"
        >
          <p
            class="text-foreground-100 font-medium text-sm underline text-ellipsis whitespace-nowrap overflow-hidden overflow-visible whitespace-pre-wrap leading-6 text-foreground-100 underline-offset-4 sm:leading-4"
          >
            Enter your Postal code for Delivery Availability
          </p>
        </a>
      </div>
    </div>
    <div
      class="relative flex gap-[14px] sm:h-[45px] before:content-[''] before:absolute before:w-full before:border-t-small before:border-background-300 before:top-[-20px]"
    >
      <div
        class="h-6 w-6 text-danger-100"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 21 21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.7812 9.1875V5.25C13.7812 4.37976 13.4355 3.54516 12.8202 2.92981C12.2048 2.31445 11.3702 1.96875 10.5 1.96875C9.62971 1.96875 8.79512 2.31445 8.17976 2.92981C7.56441 3.54516 7.21871 4.37976 7.21871 5.25V9.1875M17.1552 7.44362L18.2603 17.9436C18.3216 18.5255 17.8666 19.0312 17.2812 19.0312H3.71871C3.58064 19.0314 3.44408 19.0025 3.31791 18.9464C3.19174 18.8904 3.07877 18.8084 2.98635 18.7058C2.89393 18.6032 2.82412 18.4824 2.78145 18.3511C2.73879 18.2197 2.72422 18.0809 2.73871 17.9436L3.84471 7.44362C3.87022 7.20174 3.98438 6.97786 4.16518 6.81516C4.34597 6.65246 4.5806 6.56246 4.82383 6.5625H16.1761C16.6801 6.5625 17.1027 6.94312 17.1552 7.44362ZM7.54683 9.1875C7.54683 9.27452 7.51226 9.35798 7.45073 9.41952C7.38919 9.48106 7.30573 9.51562 7.21871 9.51562C7.13168 9.51562 7.04822 9.48106 6.98669 9.41952C6.92515 9.35798 6.89058 9.27452 6.89058 9.1875C6.89058 9.10048 6.92515 9.01702 6.98669 8.95548C7.04822 8.89394 7.13168 8.85938 7.21871 8.85938C7.30573 8.85938 7.38919 8.89394 7.45073 8.95548C7.51226 9.01702 7.54683 9.10048 7.54683 9.1875ZM14.1093 9.1875C14.1093 9.27452 14.0748 9.35798 14.0132 9.41952C13.9517 9.48106 13.8682 9.51562 13.7812 9.51562C13.6942 9.51562 13.6107 9.48106 13.5492 9.41952C13.4877 9.35798 13.4531 9.27452 13.4531 9.1875C13.4531 9.10048 13.4877 9.01702 13.5492 8.95548C13.6107 8.89394 13.6942 8.85938 13.7812 8.85938C13.8682 8.85938 13.9517 8.89394 14.0132 8.95548C14.0748 9.01702 14.1093 9.10048 14.1093 9.1875Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
      <div
        class="flex flex-col gap-[7px] overflow-hidden"
      >
        <p
          class="  no-underline text-ellipsis whitespace-nowrap overflow-hidden text-[17px] font-bold leading-5 text-primary-100"
        >
          Return Delivery
        </p>
        <p
          class="text-foreground-100 font-medium text-sm no-underline text-ellipsis whitespace-nowrap overflow-hidden overflow-visible whitespace-pre-wrap leading-6 text-foreground-100 sm:leading-4"
        >
          Free 30 days Delivery Return.Â 
          <a
            href="#"
          >
            <span
              class="underline underline-offset-4"
            >
              Details
            </span>
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
